bazel/BUILD

```diff
@@ -166,6 +166,11 @@ config_setting(
     },
 )
 
+config_setting(
+    name = "clang_pch_build",
+    values = {"define": "ENVOY_CLANG_PCH=1"},
+)
```

Comment:
use a bool_flag instead of `define`?

Reply:
I couldn't get this option to work, started a thread on it here https://envoyproxy.slack.com/archives/C7E6C71QB/p1618869455054400

Either way it should be an easy swap if we want because no one should ever use this directly, they should only ever use the `--config=clang-pch` flag instead. We might actually prefer this way because it only requires 1 target vs 2

Reply:
We talked offline and you can use `--@//` for this omitting the `envoy` part. But what do you think about merging this way since it's not meant for direct consumption anyways and this leaves us 1 less option to maintain since for the flags we have to have both.